ccl/backupccl: TestCleanupIntentsDuringBackupPerformanceRegression failed

ccl/backupccl.TestCleanupIntentsDuringBackupPerformanceRegression [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsGcpLinuxX8664BigVm_CclUnitTests/12282055?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsGcpLinuxX8664BigVm_CclUnitTests/12282055?buildTab=artifacts#/) on master @ [07915532f9a7def000127029c667c3a91ce90e47](https://github.com/cockroachdb/cockroach/commits/07915532f9a7def000127029c667c3a91ce90e47):


```
=== RUN   TestCleanupIntentsDuringBackupPerformanceRegression
    test_log_scope.go:170: test logs captured to: /artifacts/tmp/_tmp/08eca5aa191e7cb5498d48e4d60049c7/logTestCleanupIntentsDuringBackupPerformanceRegression2898872884
    test_log_scope.go:81: use -show-logs to present logs inline
    backup_intents_test.go:124: -- test log scope end --
--- FAIL: TestCleanupIntentsDuringBackupPerformanceRegression (43.72s)
test logs left over in: /artifacts/tmp/_tmp/08eca5aa191e7cb5498d48e4d60049c7/logTestCleanupIntentsDuringBackupPerformanceRegression2898872884
=== RUN   TestCleanupIntentsDuringBackupPerformanceRegression/abort=true
    backup_intents_test.go:122: 
        	Error Trace:	github.com/cockroachdb/cockroach/pkg/ccl/backupccl_test/pkg/ccl/backupccl/backup_intents_test.go:122
        	            				github.com/cockroachdb/cockroach/pkg/testutils/subtest.go:31
        	Error:      	"1100" is not greater than or equal to "1169"
        	Test:       	TestCleanupIntentsDuringBackupPerformanceRegression/abort=true
    --- FAIL: TestCleanupIntentsDuringBackupPerformanceRegression/abort=true (13.66s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
/cc @cockroachdb/disaster-recovery
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestCleanupIntentsDuringBackupPerformanceRegression.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>


Jira issue: CRDB-32609

--- a/pkg/ccl/backupccl/backup_intents_test.go
+++ b/pkg/ccl/backupccl/backup_intents_test.go
@@ -100,14 +100,13 @@ func TestCleanupIntentsDuringBackupPerformanceRegression(t *testing.T) {
 			}
 		}
 
+		// Reset the counters to avoid counting pushes and intent resolutions not
+		// part of the backup.
+		numIntentResolveBatches.Store(0)
+		numPushBatches.Store(0)
+
 		_, err = sqlDb.Exec("backup table foo to 'userfile:///test.foo'")
 		require.NoError(t, err, "Failed to run backup")
-
-		if !abort {
-			for _, tx := range transactions {
-				require.NoError(t, tx.Commit())
-			}
-		}
 
 		// We expect each group of 10 intents to take 2 intent resolution batches:
 		// - One intent gets discovered and added to the lock table, which forces the
@@ -120,5 +119,12 @@ func TestCleanupIntentsDuringBackupPerformanceRegression(t *testing.T) {
 		// Each of the 1,000 transactions is expected to get pushed once, but in an
 		// actual run of the test we might see more pushes (e.g. of other transactions).
 		require.GreaterOrEqual(t, 1100, int(numPushBatches.Load()))
+
+		if !abort {
+			for _, tx := range transactions {
+				// Ensure the long-running transactions can commit.
+				require.NoError(t, tx.Commit())
+			}
+		}
 	})
 }

